### PR TITLE
created structure for churchslavonic style and package

### PR DESCRIPTION
**Pull request by pgmmpk**
idea is:
1. Keep code in cu-num.sty, cu-util.sty, cu-kinovar.sty, cu-calendar.sty. These are **internal** packages to simplify development and testing
2. User-visible churchslavonic.sty (and gloss-churchslavonic.sty) will just pull the internal packages in
3. Get rid of cu-num/, cu-calendar/ and cu-kinovar/ sub-directories (subdirs make is harder to
   develop because of how TeX searches for files - it is easier to have all sources in current dir)
4. Documentation is in churchslavonic.tex (User Guide). Look inside to see how I am trying to
   create a bilingual doc
5. All automated tests are in tests/ subdir (not included in CTAN package)
6. Root Makefile can do "docs", "package", "test", and "clean"

### CODE REVIEW

File: cu-calendar/cu-calendar-test.tex [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+\assertSame{\cu@calendarJulian{2457491}}{2016-3-30}
+\assertSame{\cu@calendarGregorian{2457478}}{2016-3-30}
+
+\assertSame{\cu@julianFromGregorian{2016}{3}{30}}{2016-3-17}
+\assertSame{\cu@gregorianFromJulian{2016}{3}{17}}{2016-3-30}
+
+\assertSame{\cuJulianDate{2016}{3}{30}}{17 ма́рта л. 2016}
+\assertSame{\cuGregorianDate{2016}{3}{30}}{30 ма́рта л. 2016}
+
```
Comment (typiconman) [inline cu-calendar/cu-calendar-test.tex]: I think for a full test we need to test the complicated cases, for example, make sure that 5 October 1582 Julian == 15 October 1582 Gregorian. Also, what happens when a user specifies a date like 14 October 1582 Gregorian? Does this exist?
Also, check the complicated leap years, for example, make sure that 2100 _is_ a Julian leap year (29 February 2100 exists) but not a Gregorian leap year (29 February 2100 does not exist).

**Comment by typiconman** [comment]
I get a problem with make:
make: **\* [churchslavonic-en.pdf] Error 1
Log file has this:

`!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!
! fontspec error: "font-not-found"
! 
! The font "PonomarUnicode.otf" cannot be found.
! 
! See the fontspec documentation for further information.
! 
! For immediate help type H <return>.
!...............................................  
`

I think the problem is with defining the font as PonomarUnicode.otf. Where is it looking for the font file? Do we need to include the font file explicitly in the directory?

**Comment by pgmmpk** [comment]
PonomarUnicode.otf is being looked up in the TeX installation tree. I've got it in

```
/usr/local/texlive/2015basic/texmf-dist/fonts/opentype/public/churchslavonic/ponomar/
```

We can not use logical names like "Ponomar Unicode", because these are looked up in the list of
fonts installed system-wide (on the OS level). TexLive package manager does not (afaik) has a
capability to install fonts. It only downloads and copies them somewhere under textmf-dist/fonts...

Try copying font binaries to the proper place, and then re-hash tex lookup tables by running:

```
mktexlsr
```

**Comment by typiconman** [comment]
OK, I can confirm that it works after I installed the font in the TeX tree. Merging.

File: cu-calendar/cu-calendar-test.tex (hunk 1 above)
Comment (pgmmpk) [inline cu-calendar/cu-calendar-test.tex]: more tests added, see https://github.com/slavonic/cu-tex/blob/flatten-file-tree/tests/cu-calendar-test.tex#L46

note that "cuDate" just formats data, so \cuDate{2016-04-33} is perfectly acceptable and will be formatted
as April 33rd. Same with \cuDate{1582-10-14}. What happens during the conversion julian <->gregorian is
a different story. This implementation assumes that dates before split are identical.

Date conversion works in a "soft" manner, meaning that invalid input will be normalized during the conversion
(32 Jan will work as if it is 1 Feb). This needs to be documented.

Comment (typiconman) [inline cu-calendar/cu-calendar-test.tex]: LGTM.

**Pull request merged**